presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -190,7 +176,19 @@ public HivePartitionResult getPartitions(SemiTransactionalHiveMetastore metastor
         return TupleDomain.withColumnDomains(builder.build());
     }
 
-    private Optional<Map<ColumnHandle, NullableValue>> parseValuesAndFilterPartition(String partitionName, List<HiveColumnHandle> partitionColumns, List<Type> partitionTypes, Constraint<ColumnHandle> constraint)
+    private static class PartitionNameAndValues
```

Comment:
I'm not sure we need this class. We could change `parseValuesAndFilterPartition` to return `Optional<HivePartition>`. Would require passing a couple extra arguments but the code would be simpler.